### PR TITLE
Codecov 80+% check | Workflows upadte to build the whl file for spare_scores installation | More test cases and fixes

### PR DESCRIPTION
**Pull request by spirosmaggioros**
Now, the workflows are installing spare_scores directly from the build, so they can run the latest version(without updating it on PyPI). 

Added more test cases for the models(specially for exceptions)

Removed sys paths from test cases, now that the workflows build the project, test cases can run with the "spare_scores." package.

### CODE REVIEW

File: tests/unit/test_spare_scores.py [hunk 1]
```diff
@@ -1,14 +1,28 @@
-import sys
 import unittest
```
Comment (georgeSkoumas) [inline tests/unit/test_spare_scores.py:1]: Maybe time to go with pytest?

Comment (spirosmaggioros) [reply tests/unit/test_spare_scores.py:1]: pytest works, simply do:
``` cd tests/unit && pytest --cov=../../```

Comment (georgeSkoumas) [reply tests/unit/test_spare_scores.py:1]: Cool, then to make this code compatible with pytest and avoid mixing unittest syntax, could you please make the following changes:

- Remove unittest Import: We won't be using it anymore.
- Rename Test Classes: Change class names like CheckMLPDataset to TestMLPDataset. 
- Update Assertions: Replace self.assertTrue(condition) with assert condition.

Comment (spirosmaggioros) [reply tests/unit/test_spare_scores.py:1]: Ok, i will open a new issue for this and i will make the changes there. This can be merged if it's good

**Review by georgeSkoumas — APPROVED** [review]

**Pull request merged**